tests/test-api.php

```diff
@@ -584,11 +586,155 @@ public function test_client_data() {
 				'some_other_data'                => 42,
 				'donations'                      => [],
 				'user_id'                        => false,
+				'prompts'                        => [],
 			],
 			'Returns data without overwriting the existing data.'
 		);
 	}
 
+	/**
+	 * Client data rebuilding.
+	 */
+	public function test_client_data_rebuild() {
+		$api       = new Lightweight_API();
+		$client_id = 'client_' . uniqid();
+		global $wpdb;
+		$events_table_name = Segmentation::get_events_table_name();
+		$wpdb->query( $wpdb->prepare( "INSERT INTO `$events_table_name` (`type`, `client_id`, `post_id`) VALUES (%s, %s, %s)", 'post_read', $client_id, '42' ) ); // phpcs:ignore
+
+		self::assertEquals(
+			$api->get_client_data( $client_id ),
+			[
+				'suppressed_newsletter_campaign' => false,
+				'posts_read'                     => [
+					[
+						'post_id'      => '42',
+						'category_ids' => null,
+						'created_at'   => '0000-00-00 00:00:00',
+					],
+				],
+				'email_subscriptions'            => [],
+				'donations'                      => [],
+				'user_id'                        => false,
+				'prompts'                        => [],
+			],
+			'Returns expected data blueprint in absence of saved data.'
+		);
+	}
+
+	/**
+	 * Updating prompts in client data.
+	 */
+	public function test_client_data_prompts() {
+		$api       = new Lightweight_API();
+		$client_id = 'client_' . uniqid();
+		$popup_id  = Newspack_Popups_Model::canonize_popup_id( uniqid() );
+
+		// Report a prompt view.
+		self::$report_campaign_data->report_campaign(
+			[
+				'cid'      => $client_id,
+				'popup_id' => $popup_id,
+			]
+		);
+		$expected_popup_data = [
+			'count'            => 1,
+			'last_viewed'      => time(),
+			'suppress_forever' => false,
+		];
+		self::assertEquals(
+			$api->get_client_data( $client_id )['prompts'],
+			[
+				"$popup_id" => $expected_popup_data,
+			],
+			'Returns data with prompt data after a prompt is reported.'
+		);
+		self::assertEquals(
+			$api->get_campaign_data( $client_id, $popup_id ),
+			$expected_popup_data,
+			'Returns prompt data only.'
+		);
+
+		// Report another view of the same prompt.
+		self::$report_campaign_data->report_campaign(
```

Comment:
Note: changed use of `save_client_data` (more of an internal function) to `report_campaign`/`report_client_data`, so the tests are more aligned with how the code operates.

Reply:
Makes sense, thanks for clarifying!